mattermost/mattermost
# MM-52642 CLD-6352 MM-54007 Prepare setup for Playwright pipeline

#### Summary

Note:
- This change does not cover yet related to running in GHA. I'll address separately.
- Please help on testing if everything works by running the following commands:
  - `TEST=cypress make`
  - `TEST=playwright make`
  - If know the test CWS instance, may try running cloud-like test as well: `TEST=cypress SERVER=cloud make` or `TEST=playwright SERVER=cloud make` with `CWS_URL` and `MM_LICENSE` set with cloud license.
  - May run in combination with env variable `ENABLED_DOCKER_SERVICES`
    - possible values: `postgres inbucket minio openldap elasticsearch keycloak`
    - required values if overriden: `postgres inbucket`

General changes in `e2e-tests` scripts:
- Introduced `ENABLED_DOCKER_SERVICES` for selective running of services. Helps with targeted testing in local development for lightweight and faster running of docker services. Ability to trim down instead of running full dockers.
- Introduced `TEST` with possible value of `cypress` (default), `playwright` or `server`.
- Introduced `SERVER` with possible value of `self-hosted` (default) or `cloud`.
- Used of Elasticsearch' arm64 to match host machine.
- Added options to run in cloud-like test server with communication to test CWS by setting up `CWS_URL` env.
- Changed `env` to `.env` and not save `MM_LICENSE` to `.env.server` to prevent from accidentally pushing unintended change.
- Removed `utils` container (`mattermost/mattermost-build-server`) initially set for overriding default server config. It's no longer required since most if not all are passed as env variable to container, and others are handled by the test framework.
- Run Cypress either with `TEST=cypress make` or simply `make` for self-hosted test server (default) or `TEST=cypress SERVER=cloud make` for cloud-like test server.
- Run Playwright either with `TEST=playwright make` for self-hosted test server (default) or `TEST=playwright SERVER=cloud make` for cloud-like test server.
- Added `make fmt-ci` to standardize format and check to yaml files and shell scripts with `prettier`, `shellcheck` and `shellfmt`.

Playwright test application changes:
- Bumped Playwright to `1.38.1`
- Updated test snapshots (screenshots)
- Fixed tests, one related to accessibility on sidebar plugin icon
- Removed hanging around mobile-related tests (as part of deprecation support on mobile view)
- Reorganized types

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-52642
- https://mattermost.atlassian.net/browse/CLD-6332
- https://mattermost.atlassian.net/browse/MM-54007

#### Release Note
```release-note
NONE
```
